zio-http/src/main/scala/zio/http/codec/HttpCodec.scala

```diff
@@ -75,15 +75,28 @@ sealed trait HttpCodec[-AtomTypes, Value] {
     HttpCodec.Combine[AtomTypes1, AtomTypes1, Value, Value2, combiner.Out](self, that, combiner)
 
   /**
-   * Combines two query codecs into another query codec.
+   * To end the route codec and begin with query codec, and re-interprets as
+   * PathQueryCodec
+   */
+  final def :?[Value2](
+    that: QueryCodec[Value2],
+  )(implicit
+    combiner: Combiner[Value, Value2],
+    ev: HttpCodecType.Path <:< AtomTypes,
+  ): HttpCodec[HttpCodecType.PathQuery, combiner.Out] =
+    (self ++ that).asInstanceOf[HttpCodec[HttpCodecType.PathQuery, combiner.Out]]
+
+  /**
+   * Append more query parameters to either a query codec, or to a pathQuery
+   * codec which is a combination of path and query
    */
   final def &[Value2](
     that: QueryCodec[Value2],
   )(implicit
     combiner: Combiner[Value, Value2],
-    ev: HttpCodecType.Query <:< AtomTypes,
-  ): QueryCodec[combiner.Out] =
-    self.asQuery ++ that
+    ev: HttpCodecType.PathQuery with HttpCodecType.Query <:< AtomTypes,
+  ): HttpCodec[AtomTypes, combiner.Out] =
```

Comment:
You need:

1. Implicit evidence that `HttpCodecType.Query with HttpCodecType.Path <:< AtomTypes`
2. Return `HttpCodec[AtomTypes with HttpCodecType.Query, combiner.Out]`

Reply:
@jdegoes  The problem with the evidence of `HttpCodecType.Query with HttpCodecType.Path <:< AtomTypes` is that, it will allow ` "accounts"  / "customer" & paramStr("id")` which is wrong,

Reply:
The reason for that is , the `AtomTypes` of  `"accounts"  / "customer"` is `Path` and `HttpCodecType.Query with HttpCodecType.Path <:< AtomTypes` satisfies it. This is the reason for introducing a fully separate type called `PathQuery` .

`HttpCodecType.PathQuery with HttpCodecType.Query <:< AtomTypes` implies: in order to call `&`, it has to be either a `path-appended-with-atleast-one-query-parameter`, or it can be just a `Query` (example: `paramStr("id") & paramStr("age")`)